mobile/src/store/courses/actions.ts

```diff
@@ -75,19 +75,35 @@ const getMentorsByCourseId = createAsyncThunk<
 
 const updateVisibilityBecomeMentor = createAsyncThunk<
   boolean,
-  void,
+  number,
   AsyncThunkConfig
->(ActionType.UPDATE_VISIBILITY_BECOME_MENTOR, async (_, { getState }) => {
-  const {
-    auth: { user },
-    courses: { mentors, course },
-  } = getState();
-  const isMentorBecomingVisible =
-    course?.courseCategoryId &&
-    !mentors.some((mentor) => mentor.id === user?.id);
+>(
+  ActionType.UPDATE_VISIBILITY_BECOME_MENTOR,
+  async (userId, { extra, getState }) => {
+    const {
+      courses: { course },
+    } = getState();
 
-  return Boolean(isMentorBecomingVisible);
-});
+    const { coursesApi, interviewsApi } = extra;
+
+    const isMentor =
```

Comment:
Variables that starts with is or has should be only boolean

Reply:
But `coursesApi.checkIsMentor` returns boolean

Reply:
If we should have course on this step, then use `as` operator, if not - check if there is a course and return from the function if not
```
  const isMentor = await coursesApi.checkIsMentor({
        courseId: (course as CourseDto).id,
  }));
```